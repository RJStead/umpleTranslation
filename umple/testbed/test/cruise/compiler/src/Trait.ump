generate Java "../src-gen-umple";

class Color {}

trait TDrawable {
  Color color;
  public void foo() {
    if (color != null) {
        color = null;
    }
    return color;
  }
}

class Z {
  isA TDrawable;
}
