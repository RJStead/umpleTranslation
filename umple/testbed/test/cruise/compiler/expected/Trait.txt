Trait.ump:11: error: incompatible types: unexpected return value
    return color;
           ^
1 error